rholang/src/test/scala/coop/rchain/rholang/interpreter/accounting/CostAccountingSpec.scala

```diff
@@ -63,4 +78,109 @@ class CostAccountingSpec extends FlatSpec with TripleEqualsSupport {
 
     assertOutOfPhloError(test)
   }
+
+}
+
+class CostAccountingSpec extends FlatSpec with Matchers with PropertyChecks {
+
+  private[this] def evaluateWithCostLog(
+      initialPhlo: Long,
+      contract: String
+  ) = {
+    val dbDir                              = Files.createTempDirectory("cost-accounting-spec-")
+    val size                               = 1024L * 1024 * 1024
+    implicit val errorLog                  = new ErrorLog[Task]()
+    implicit val logF: Log[Task]           = new Log.NOPLog[Task]
+    implicit val metricsEff: Metrics[Task] = new metrics.Metrics.MetricsNOP[Task]
+
+    (for {
+      costAlg <- CostAccounting.of[Task](Cost(initialPhlo))
+      costL   <- costLog[Task]
+      cost    = loggingCost(costAlg, costL)
+      costsLoggingProgram <- {
+        costL.listen({
+          implicit val ca = costAlg
+          implicit val c  = cost
+          for {
+            runtime <- Runtime
+                        .create[Task, Task.Par](dbDir, size, StoreType.LMDB)
+            par <- Interpreter[Task].buildNormalizedTerm(contract)
+            res <- Interpreter[Task]
+                    .evaluate(runtime, par, Cost(initialPhlo.toLong))
+            _ <- runtime.close()
+            _ <- Task.eval(dbDir.recursivelyDelete())
```

Comment:
It has now occurred to me, that we might be testing a code path that's not relevant to production, simply because the production uses `RuntimeManager.injAttempt`. Especially that I remember us discussing problems with reproducing the right behaviour using Runtime and Reducer.inj. I think what we should do, is expose `injAttempt` and use it here. I'd hate this to become a case of '2 unit tests, 0 integration tests'.

Reply:
Your concern shows an issue with the possibility of the RuntimeManager and Interpreter functionalities not being aligned. The correct solution of this problem would be to extract the common functionality of both and remove all doubt, but since this is only loosely related to testing of cost accounting I would suggest doing this in a separate PR as this one is definitely going to bloat.

Reply:
Fine. Just make sure that happens before you stop working on those tests.

I've got to tell you I'm already very wary of the 'let's do that in a following PR mantra'. Since we're already going to have a follow-up for this once @JosDenmark's work lands, so be it.

Reply:
Please also mind that 'correct solution of this problem would be to extract the common functionality of both and remove all doubt' would mean 'remove all traces of injAttempt from RuntimeManager'. Otherwsie we need to test via injAttempt.

Reply:
The "let's do that in a following PR mantra" is inevitable when working in areas of code with significant amount of debt. When the true fix for the newly uncovered and loosely related problem might cause the PR to bloat I would rather have it implemented separately instead of mixing concerns. This is only an opinion though, yours might be different.